cpp/ps0004/ps0004_sol.cpp

```diff
@@ -0,0 +1,25 @@
+#include <iostream>
+
+int main()
+{
+    // prompting user for number 1/1.5
+    float number;
+    std::cout<<"Enter the number from 32 to 126\n";
+    std::cin>>number;
+    int temp_number = (int)number;
+    char opposite_characer = temp_number;
+    
+    // checking invalidaty of user's input 2/4
+    // printing the number and the opposite printable character 1/2
+    if(number==0)
+        std::cout<<"invalid character/s or zero value \n";
+    else if(number!=temp_number)
+        std::cout<<"invalid a floating point value\n";
+    else if(number>=32 && number<=126)
+        std::cout<<"the character for "<<number<<"is"<<opposite_characer<<"\n";
+    else
+        std::cout<<"invalid number out of the range\n";
```

Comment:
- [ ] used `if` structure in a right way
- [ ] `if` structure take in consideration all invalid cases
- [ ] all expressions to check invalidity are correct
- [x] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [ ] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character